.github/workflows/publish.yml

```diff
@@ -0,0 +1,20 @@
+name: Publish charm
+
+on:
+  workflow_dispatch:
```

Comment:
Right now, the publish step is only occurring when the workflow is dispatched, but I'm not seeing where the workflow would be dispatched (e.g. invoked from another workflow).

To configure this to publish automatically to the `latest/edge` channel on merge into the main branch, you can configure this as:
```yaml
on:
  push:
    branches:
      - main
```